### PR TITLE
[dbnode] Fix concurrency granularity of seekerManager.UpdateOpenLease

### PR DESCRIPTION
**Pull request by linasm**
**What this PR does / why we need it**:
`seekerManager.UpdateOpenLease(...)` was preventing concurrent calls at global granularity (see the removed comment). This changes it to the smallest granularity possible (namespace + shard + block start) to allow concurrent operations.
Such change would prevent possible collisions between `shard.AggregateTiles()` background running and manual invocations, and `shardColdFlush.Done()`.

**Special notes for your reviewer**:
Noticed this while investigating a flaky integration test.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
NONE

**Does this PR require updating code package or user-facing documentation?**:
NONE

### CODE REVIEW

**Review by notbdu — APPROVED** [review]
LGTM

File: src/dbnode/persist/fs/seek_manager.go [hunk 1]
```diff
@@ -91,9 +91,9 @@ type seekerManager struct {
 	bytesPool      pool.CheckedBytesPool
 	filePathPrefix string
 
-	status          seekerManagerStatus
-	shardSet        sharding.ShardSet
-	isUpdatingLease bool
+	status                     seekerManagerStatus
+	shardSet                   sharding.ShardSet
+	updateOpenLeasesInProgress map[block.HashableLeaseDescriptor]struct{}
```
Comment (notbdu) [inline src/dbnode/persist/fs/seek_manager.go:96]: +1 on consistent naming w/ what's in `block/lease.go`

File: src/dbnode/persist/fs/seek_manager.go [hunk 2]
```diff
@@ -500,25 +503,26 @@ func (m *seekerManager) UpdateOpenLease(
 	return updateLeaseResult, nil
 }
 
-func (m *seekerManager) startUpdateOpenLease(descriptor block.LeaseDescriptor) (bool, error) {
+func (m *seekerManager) startUpdateOpenLease(
+	namespace ident.ID,
+	hashableDescriptor block.HashableLeaseDescriptor,
+) (bool, error) {
 	m.Lock()
 	defer m.Unlock()
```
Comment (notbdu) [inline src/dbnode/persist/fs/seek_manager.go:511]: nit: would we want to release this lock and perhaps use a `sync.Map` like we do in `block/lease.go` for tracking open leases? Although tbh perf wise this prob wouldn't make a diff at all.

Comment (linasm) [reply src/dbnode/persist/fs/seek_manager.go:511]: Using `sync.Map` (as I did in `block/lease.go`) was the initial approach that I tried, but then I realized that the lock is being held for the whole duration of `startUpdateOpenLease`, and there is this `namespace` check in between progress check / update, so I decided to go with a simple `map` under this lock (to better preserve the original semantics).

**Pull request merged**